docs/elasticsearch.txt

```diff
@@ -84,6 +84,73 @@ JanusGraph only uses default values for `index-name` and `health-request-timeout
+
+In addition to enabling SSL support, you may use non-default location for SSL Trust Store - for example if you run your own CA etc. This can be done as follows:
+
+[source, properties]
+----
+index.search.elasticsearch.ssl.truststore.location=/path/to/your/truststore.jks
+index.search.elasticsearch.ssl.truststore.password=truststorepwd
+----
+
+Note that these settings apply only to Elasticsearch REST client and do not affect any other SSL connections in JanusGraph.
+
+===== SSL client authentication for HTTPS
+
+Regardless of HTTP authentication settings, the client may be configured to use SSL client authentication.
+
+[source, properties]
+----
+index.search.elasticsearch.ssl.keystore.location=/path/to/your/keystore.jks
+index.search.elasticsearch.ssl.keystore.storepassword=keystorepwd
+index.search.elasticsearch.ssl.keystore.keypassword=keypwd
+----
+
+Any of the passwords can be empty.
+
+
+===== REST Client HTTP Authentication
+
+REST client supports the following authentication options: Basic HTTP Authentication (username/password) and custom authentication based on the user-provided implementation.
+
+These authentication methods are independent from SSL client authentication described above.
+
+====== Basic HTTP Authentication
+
+Basic HTTP Authentication is available regardless of the state of SSL support.  Authentication realm value is optional. If it is not provided, the credentials will match any authentication realm value.
+
+
+[source, properties]
+----
+index.search.elasticsearch.http.auth.type=basic
+index.search.elasticsearch.http.auth.basic.username=httpuser
+index.search.elasticsearch.http.auth.basic.password=httppassword
+index.search.elasticsearch.http.auth.basic.realm=myesrealm
+----
+
+[TIP]
+It is highly recommended to use SSL (e.g. setting `index.[X].elasticsearch.ssl.enabled` to `true`) when using this option as the credentials can be intercepted when sent over an unencrypted connection!
+
+====== Custom HTTP Authentication
+
+Additional authentication methods can be implemented by providing your own implementation. The custom authenticator is configured as follows:
+
+[source, properties]
+----
+index.search.elasticsearch.http.auth.custom.authenticator-class=fully.qualified.class.Name
+index.search.elasticsearch.elasticsearch.http.auth.custom.authenticator-args=arg1,arg2,...
+----
+
+Argument list is optional and can be empty.
+
+The class specified there has to implement `org.janusgraph.diskstorage.es.rest.util.RestClientAuthenticator` interface or extend `org.janusgraph.diskstorage.es.rest.util.RestClientAuthenticatorBase` convenience class. The implementation gets access to HTTP client configuration and can customize the client as needed. Refer to <<javadoc>> for more information.
+
+You can see an example of a custom authenticator for Elasticsearch REST client in <<escustomauth>>.
```

Comment:
I think you could put the AWS example here instead of in the appendix.

Reply:
done